components/GoalListTabEdit.vue

```diff
@@ -1,6 +1,8 @@
 <template>
   <tr v-if="goal">
-    <td @click="openCreateGoalModal(state.id)">{{ state.title }}</td>
+    <nuxt-link :to="`/goal/${props.goalId}`">
```

Comment:
while navigating here to the modal, can we pass it a reference to the current url we are at, so on close it sends us back to it

Reply:
We can use router.back() from useRouter composable for that. Need to have a discussion regarding this tho.

Reference - https://nuxt.com/docs/api/composables/use-router